File: Python/Algorithms/CountingSort.py
def countingSort(myarr):
    temparr =  [0 for _ in range(max(myarr)+1)]
    for val in myarr:
        temparr[val]+=1
    
    for i in range(1,max(myarr)+1):
        temparr[i]+=temparr[i-1]

    for i in range(max(myarr) , 0, -1):
        temparr[i] = temparr[i-1]
    
    temparr[0] = 0
    valuearr = [0 for _ in range(len(myarr))]
    for val in myarr:
        valuearr[temparr[val]] = val
        temparr[val]+=1

    for i in range(len(myarr)):
        myarr[i] = valuearr[i]

arr = [9,2,5,2,7,5]
countingSort(arr)
print(*arr)

File: C++/Algorithms/CountingSort.cpp
#include<iostream>
using namespace std;

void countingSort(int myarr[] , int size){
    
    int maxvalue = myarr[0];
    for(int i = 1 ; i <size;i++){
        if (maxvalue<myarr[i]) maxvalue=myarr[i];
    }

    int temparr[maxvalue];
    for(int i =0;i<maxvalue+1;i++){
        temparr[i] = 0;
    }

    for(int i = 0 ;i<size;i++){
        temparr[myarr[i]]+=1;
    }

    for(int i = 1; i < maxvalue+1;i++){
        temparr[i]+= temparr[i-1];
    }

    for(int i = maxvalue ; i>0;i--){
        temparr[i] = temparr[i-1];
    }
    temparr[0] = 0 ;

  

    int valuearr[size];
    for(int i = 0 ;i<size;i++){
        valuearr[temparr[myarr[i]]] = myarr[i];
        temparr[myarr[i]]+=1;
    }


    for(int i = 0 ;i<size;i++){
        myarr[i] = valuearr[i];
    }



}


int main(){

    int arr[] = {9,2,5,2,7,5};


    int size = sizeof(arr)/sizeof(arr[0]);
    countingSort(arr, size);

    for(int i =0 ;i<size ;i++){
        cout<<arr[i]<<" ";
    }

    return 0;
}